src/modules/vtol_att_control/tailsitter.cpp

```diff
@@ -294,12 +301,27 @@ void Tailsitter::fill_actuator_outputs()
 			_torque_setpoint_0->xyz[2] = _vehicle_torque_setpoint_virtual_fw->xyz[2] * _param_vt_fw_difthr_s_r.get();
 		}
 
+		// for the short period after switching to FW where there is no thrust published yet from the FW controller,
+		// keep publishing the last MC thrust to keep the motors running
+		if (hrt_elapsed_time(&_trans_finished_ts) < 50_ms) {
```

Comment:
Instead of hard coding these little magic timeouts everywhere what about checking again the actual FW controller publication (_fw_virtual_att_sp)?

Reply:
I didn't find a good way without the hard coded timeout. 
- `_fw_virtual_att_sp` is constantly being published in the transition phase (with thrust=0), so barely looking at the timestamp doesn't help. We would need to do some isNone() checks.
- We would need to be careful between the `_fw_virtual_att_sp` publication and the torque/thrust handling we do here, as one comes from the position controller while the other is the output from the FW rate controller.